pkg/pillar/nireconciler/linux_config.go

```diff
@@ -636,6 +636,40 @@ func (r *LinuxNIReconciler) getIntendedNIL3Cfg(niID uuid.UUID) dg.Graph {
 			}, nil)
 		}
 	}
+	// Add statically defined routes into the NI routing table.
+	// No need to do this if GW is one of the apps and traffic from one app
+	// to another is just forwarded by the host, not routed.
+	for _, route := range ni.config.StaticRoutes {
+		isAppGW := r.getNISubnet(ni).Contains(route.Gateway)
+		if isAppGW && r.disableAllOnesNetmask {
+			// Route is not needed inside the host, traffic is just forwarded
+			// by the bridge.
+			continue
+		}
+		if !isAppGW && !r.routeGwIsConnected(route, ni.bridge.Uplink) {
+			// GW is not routable with the current uplink.
+			continue
+		}
+		family := netlink.FAMILY_V4
+		if route.Gateway.To4() == nil {
+			family = netlink.FAMILY_V6
+		}
+		outputIf := linux.RouteOutIf{UplinkIfName: uplink}
+		if isAppGW {
+			outputIf = linux.RouteOutIf{BridgeIfName: ni.brIfName}
+		}
+		intendedL3Cfg.PutItem(linux.Route{
```

Comment:
do we check here to prevent the app's static route is 0/0 to be installed into host side?

Reply:
Currently there is no such check. How should we handle the case when NI uplink has default route and at the same time user configures static default route (with possibly different GW IP)? Which one should take precedence? I would think that the static route, since it would give the user the option to override the default route received from an external DHCP server.

Reply:
Yeh, i would think the user explicit one should take precedence.

Reply:
Done, PR is updated - user-configured default route will override the default route of the uplink.